Flaky `test_accept_invalid_certificate/test.py::test_strict_reject_with_config`

[CI](https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=master&sha=ac7430afdb07f2ce1c4d295d30e814b3cb5af1d0&name_0=MasterCI&name_1=Integration%20tests%20%28amd_asan%2C%20db%20disk%2C%20old%20analyzer%2C%203%2F6%29&name_1=Integration%20tests%20%28amd_asan%2C%20db%20disk%2C%20old%20analyzer%2C%203%2F6%29), [CIDB](https://play.clickhouse.com/play?user=play&run=1#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkoaGVhZF9yZWYpLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2FjY2VwdF9pbnZhbGlkX2NlcnRpZmljYXRlL3Rlc3QucHk6OnRlc3Rfc3RyaWN0X3JlamVjdF93aXRoX2NvbmZpZycKICAgIC0tIEFORCBjaGVja19uYW1lID0gJ0ludGVncmF0aW9uIHRlc3RzIChhbWRfYXNhbiwgZGIgZGlzaywgb2xkIGFuYWx5emVyLCA0LzYpJwogICAgQU5EIHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicpCiAgICBBTkQgKChoZWFkX3JlZiA9ICdtYXN0ZXInIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCkpCkdST1VQIEJZIGRheQpPUkRFUiBCWSBkYXkgREVTQwo=)

### Error
```py
File: test_accept_invalid_certificate/test.py:118 - in test_strict_reject_with_config
    assert "alert certificate required" in str(err.value)
E   AssertionError: assert 'alert certificate required' in 'Client failed! Return code: 210, stderr: Code: 210. DB::NetException: SSL Exception: error:80000068:system library::Connection reset by peer, while writing to socket (172.16.5.1:56952 -> 172.16.5.7:9440): (172.16.5.7:9440, local address: 172.16.5.1:56952). (NETWORK_ERROR)\n\n'
E    +  where 'Client failed! Return code: 210, stderr: Code: 210. DB::NetException: SSL Exception: error:80000068:system library::Connection reset by peer, while writing to socket (172.16.5.1:56952 -> 172.16.5.7:9440): (172.16.5.7:9440, local address: 172.16.5.1:56952). (NETWORK_ERROR)\n\n' = str(QueryRuntimeException('Client failed! Return code: 210, stderr: Code: 210. DB::NetException: SSL Exception: error:80000068:system library::Connection reset by peer, while writing to socket (172.16.5.1:56952 -> 172.16.5.7:9440): (172.16.5.7:9440, local address: 172.16.5.1:56952). (NETWORK_ERROR)\n\n'))
E    +    where QueryRuntimeException('Client failed! Return code: 210, stderr: Code: 210. DB::NetException: SSL Exception: error:80000068:system library::Connection reset by peer, while writing to socket (172.16.5.1:56952 -> 172.16.5.7:9440): (172.16.5.7:9440, local address: 172.16.5.1:56952). (NETWORK_ERROR)\n\n') = <ExceptionInfo QueryRuntimeException('Client failed! Return code: 210, stderr: Code: 210. DB::NetException: SSL Exception: error:8000...socket (172.16.5.1:56952 -> 172.16.5.7:9440): (172.16.5.7:9440, local address: 172.16.5.1:56952). (NETWORK_ERROR)\n\n') tblen=5>.value
```


### Misc
Similar to https://github.com/ClickHouse/ClickHouse/issues/67912, but not the same

--- a/tests/integration/test_accept_invalid_certificate/test.py
+++ b/tests/integration/test_accept_invalid_certificate/test.py
@@ -115,7 +115,12 @@ def test_strict_reject():
 def test_strict_reject_with_config():
     with pytest.raises(Exception) as err:
         execute_query_native(node1, "SELECT 1", config_accept)
-    assert "alert certificate required" in str(err.value)
+    # Accept both error messages due to race condition in SSL handshake:
+    # - "alert certificate required": TCP layer transmits Alert before close() executes
+    # - "Connection reset by peer": close() executes before TCP layer transmits Alert from send buffer
+    # Race condition: send() is async (returns after copying to kernel buffer), close() may execute
+    # before TCP layer actually sends the Alert packet, causing RST to be sent instead
+    assert "alert certificate required" in str(err.value) or "Connection reset by peer" in str(err.value)
 
 
 def test_strict_connection_reject():

